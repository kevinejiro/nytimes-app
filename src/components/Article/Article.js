import React, { memo } from 'react';
import { Media } from 'reactstrap';
import { Link } from 'react-router-dom';

import './Article.scss'

/**
 *
 * @param {*} props
 * @returns {JSX} JSX element
 */
function ArticleComponent(props) {
  return (
		<Link to={`/${props.id}`}>
			<Media className="article-component">
				<Media left>
					<img src={props.media[0]['media-metadata'][1].url} alt="airticleImage" ></img>
				</Media>
				<Media body>
					<Media heading>
						{props.title}
					</Media>
					<div className="by-line" >{props.byline}</div>
					<div className="date" >
						<svg
							xmlns="http://www.w3.org/2000/svg"
							width="15"
							height="15"
							viewBox="0 0 25 25"
							>
							<path d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z" />
							<path fill="none" d="M0 0h24v24H0z" />
						</svg>
						{props.published_date}
					</div> 
				</Media>
				<div className="right-svg">
					<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24">
						<path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" />
						<path d="M0 0h24v24H0z" fill="none" />
					</svg>
				</div>
			</Media>
		</Link>
  );
};

export const Article = memo(ArticleComponent);
