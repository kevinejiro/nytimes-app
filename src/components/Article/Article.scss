.article-component.media {
	padding: 10px;
	background-color: #eeeeee;

	&:hover {
		background-color: #ddd;
		transition-delay: 0.1s;
	}
}

.media-left {
	align-self: center;

	img {
		border-radius: 100%;
		height: 50px;
		object-fit: cover;
	}
}

.date {
	float: right;
	display: flex;
	color: #666666;
	font-size: 14px;
	margin-left: 0.3em;

	svg {
		margin-right: 0.3em;
		margin-top: 0.2em;
		fill: #666666;
	}
}

.media-body {
	padding: 10px;
}


.right-svg {
	align-self: center;
	padding-right: 0.1em;
	padding-left: 0.7em;
	display: flex;

	svg {
		fill: #666666;
	}
}

.media-heading {
	font-size: 15px;
	font-weight: bold;
	color: #222222;
}

.by-line {
	color: #666666;
	display: inline;
	font-size: 14px;
}
