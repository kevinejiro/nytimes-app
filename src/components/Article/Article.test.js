import React from 'react';
import { shallow } from 'enzyme';

import { Article } from './Article';

it('renders without crashing', () => {
  const component = <Article />;
  const wrapper = shallow(<div>{component}</div>)
  expect(wrapper).toMatchSnapshot();
  }); 