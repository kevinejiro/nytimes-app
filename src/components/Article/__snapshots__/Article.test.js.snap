// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <UNDEFINED />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UNDEFINED />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UNDEFINED />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
